iwyu_ast_util.h

```diff
@@ -548,6 +549,11 @@ clang::SourceRange GetSourceRangeOfClassDecl(const clang::Decl* decl);
 set<clang::FunctionDecl*> GetLateParsedFunctionDecls(
     clang::TranslationUnitDecl* decl);
 
+struct TemplateInstantiationData {
```

Comment:
Could you add two constructors to this type; one for the resugar map and one for resugar map + provided types? I find the initialization syntax currently used to construct it is a little too implicit.

Reply:
Wouldn't the constructor calling syntax look the same? Eventually, we can turn on `--std=c++20` and use [designated initializers](https://en.cppreference.com/w/cpp/language/aggregate_initialization#Designated_initializers).

Reply:
> Wouldn't the constructor calling syntax look the same?

I mean that constructors can equally be called with curly braced syntax as well as with parentheses. But if you prefer parentheses, I can add constructors, of course.

Reply:
I want to make sure the type name is spelled out when constructing it; `TemplateInstantiationData(resugar_map)` instead of `{resugar_map}`

Reply:
Hmm, maybe it's possible to do `TemplateInstantiationData{resugar_map}` even without explicit constructors?

Reply:
Yes, it is.

Reply:
Ah, cool. Yeah, my main priority is that the type is visible where used. I still rely 100% on `grep` for navigating code, so I'm a big fan of type names :)

Reply:
Btw, I've noticed recently that writing
```cpp
for (const std::pair<KeyT, ValueT>& item : map)
```
leads to silent data copying because `std::map::value_type` is std::pair<**const** KeyT, ValueT>. So, I prefer automatic type deduction.